### DISCUSSION TITLE
A licensing question

### DISCUSSION THREAD

**Discussion by vanta240i** [mdbootstrap/TW-Elements#1783] [Support from community]
I am so confused by the AGPL license on this software. If I use this software as part of my app to allow it to select a time, do I have to make my app open source as well? I need to sell this software commercially and it MUST be closed source.

**Comment by filipkappa** [comment]
The plan with Tailwind Elements is to give as much as we can to the open-source community for free.

We believe that **AGPL** is more **in line with the open-source spirit**, which is focused on **sharing knowledge** and driving innovation.

Of course, we understand that the AGPL **may not be the best fit for some specific use cases**, which is why we have used a **dual licensing model**. This allows us to offer Tailwind Elements under the AGPL for those who share our commitment to open-source values, while also offering a custom license for **those who need a more permissive license for their specific commercial use case**.

It enables us to keep in touch with the people who required a different license, learn what kind of projects and use cases they have and adjust the licensing to it.

TE is still in beta, so our main goal, for now, is to discover the needs & requirements for such a package in depth. It's meant to be a community-driven project, and we hope this decision will help to enrich the wider open-source community as well, by requiring most of non-sensitive innovations and implementations to be shared.

If you need a different licensing option, we encourage you to **reach out to tailwind@mdbootstrap.com** and describe the needs of your project ;)
